lib/src/utils/block.dart

```diff
@@ -163,36 +163,35 @@ class BlockUtils {
   }
 
   static Future<bool> _setHashAndSignature(
-      AccountBlockTemplate transaction, KeyPair currentKeyPair) async {
-    transaction.hash = BlockUtils.getTransactionHash(transaction);
-    var transSig =
-        await BlockUtils._getTransactionSignature(currentKeyPair, transaction);
-    transaction.signature = transSig;
+      AccountBlockTemplate transaction, Signer signer) async {
+    var source = BlockUtils.getTransactionBytes(transaction);
+    transaction.hash = Hash.digest(source);
```

Comment:
```suggestion
    transaction.hash = BlockUtils.getTransactionHash(transaction);
```
This could be restored to how it was originally. And I think the `BlockUtils.getTransactionBytes` is now an unused function?

Reply:
I agree, and not really. The `getTransactionHash` method uses it.

`return Hash.digest(getTransactionBytes(transaction));`

Reply:
Yes because it was changed to work that way in this PR. But currently that change serves no purpose, since `getTransactionBytes` isn't needed anywhere else.

Obviously it doesn't really matter if it's changed to that but from a principles point of view, I prefer keeping changes as minimal as possible compared to the target branch.

In any case, this PR looks good so I'm approving it.

Reply:
I get what you're saying, but while the SDK doesn't use the method directly, it's actually quite convenient to keep them separate so that the ledger implementation can re-use it.

Although one might argue that you should isolate both implementations. It that case I would have to copy the method in the ledger impl.

Reply:
Okay I didn't realize the ledger implementation uses it. All good then.